### PR TITLE
feat: tutorial renders mdx content

### PR DESCRIPTION
**Pull request by kendallstrautman**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-kstutorial-page-skeleton-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-intro) 🔎
- [Asana task](https://app.asana.com/0/home/1199634971443477/1201957666036475) 🎟️

## What

This PR adds serializes mdx content and renders it properly for the tutorial view. A few custom components are still missing, but we will add these one by one once they are ready for dev from design. 

## Why

To make this view a bit prettier 💅 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201957666036475

### CODE REVIEW

File: src/lib/remark-plugins/rewrite-static-assets.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/**
+ * This is based on mktg-content-workflow's docs transformers
+ * https://github.com/hashicorp/mktg-content-workflows/blob/main/shared/transforms/rewrite-static-assets.ts
+ *
+ * It rewrites image paths to source from the tutorials repository
+ * via the GitHub CDN, which is provided by the assets API in mtkg-content-workflows
+ */
```
Comment (kendallstrautman) [inline src/lib/remark-plugins/rewrite-static-assets.ts:7]: Copied this directly [from Learn](https://github.com/hashicorp/learn/blob/master/lib/remark-plugins/rewrite-static-assets.ts)

File: src/views/tutorial-view/utils/get-video-url.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { VideoHostOption } from 'lib/learn-client/types'
+
+const urlBase = {
```
Comment (kendallstrautman) [inline src/views/tutorial-view/utils/get-video-url.ts:3]: This is used in the `serializeContent` function, [copied from Learn](https://github.com/hashicorp/learn/blob/master/lib/get-video-url.ts)

File: src/views/tutorial-view/utils/mdx-components.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import codeBlockPrimitives from '@hashicorp/react-code-block/mdx'
+import Tabs, { Tab } from '@hashicorp/react-tabs' // @TODO use the new tabs within dev-portal, once we figure out groups
+/**
+ * @TODO move over these components from learn, update to new spec
+ *
+ * import Accordion from 'components/accordion'
```
Comment (kendallstrautman) [inline src/views/tutorial-view/utils/mdx-components.tsx:6]: Not copying these components over for now. We'll do them one by one once they're ready for dev.

File: src/views/tutorial-view/utils/serialize-content.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import { serialize } from 'next-mdx-remote/serialize'
```
Comment (kendallstrautman) [inline src/views/tutorial-view/utils/serialize-content.ts:1]: [Copied from Learn](https://github.com/hashicorp/learn/blob/master/lib/data-fetching/tutorial-page/serialize-content.ts)

File: src/pages/waypoint/tutorials/[...tutorialSlug]/index.tsx [hunk 5]
```diff
@@ -7,7 +7,7 @@ import {
   TutorialPagePaths,
```
Comment (ashleemboyer) [inline src/pages/waypoint/tutorials/[...tutorialSlug]/index.tsx:7]: What prompted these changes? No issues, purely curious. 😊

File: src/views/tutorial-view/index.tsx [hunk 6]
```diff
@@ -1,13 +1,23 @@
+import Content from '@hashicorp/react-content'
```
Comment (ashleemboyer) [inline src/views/tutorial-view/index.tsx:1]: 💭 I didn't know this existed! I think we need to get one of these for DevDot... we're basically doing the same thing in `SidebarSidecarLayout` for applying special styles.

For more context, some of the components we're passing into `components` are: `Heading`, `InlineLink`, and `Text`. You can see how they're used for docs MDX here: https://github.com/hashicorp/dev-portal/blob/assembly-ui-v1/src/layouts/sidebar-sidecar/utils/_local_platform-docs-mdx/index.jsx.

**Review by ashleemboyer — APPROVED** [review]
Same note here as #173, might want to wait to merge so that the pages in this PR can be updated to use `CoreDevDotLayout`?

**Review by zchsh — APPROVED** [review]
💅

File: src/pages/waypoint/tutorials/[...tutorialSlug]/index.tsx (hunk 5 above)
Comment (kendallstrautman) [reply src/pages/waypoint/tutorials/[...tutorialSlug]/index.tsx:7]: I think I got some warning or issue re: destructuring on an import!

File: src/views/tutorial-view/index.tsx (hunk 6 above)
Comment (kendallstrautman) [reply src/views/tutorial-view/index.tsx:1]: Oh interesting! Y'all aren't using the react-content component? Is there a different component I should use to get the dev dot styles?

File: src/pages/waypoint/tutorials/[...tutorialSlug]/index.tsx (hunk 5 above)
Comment (ashleemboyer) [reply src/pages/waypoint/tutorials/[...tutorialSlug]/index.tsx:7]: Ah okay, that makes sense!

**Pull request merged**